src/util/TimeUtil.js

```diff
@@ -102,14 +102,18 @@ export default {
    *
    * @param {Date} date The Date object for the grace period expiry
    * @param {LanguageCode} languageCode The user's language code / locale
+   * @param {RoundDownToNearestHour} boolean Whether to round down to nearest hour. Defaults to true
    * @return {string} The formatted `short-with-timezone` local string
    */
-  formatDateToDeviceAssuranceGracePeriodExpiryLocaleString: (date, languageCode) => {
+  formatDateToDeviceAssuranceGracePeriodExpiryLocaleString: (date, languageCode, roundDownToNearestHour = true) => {
     try {
     // Invalid Date objects will return NaN for valueOf()
       if (date && !isNaN(date.valueOf()) && languageCode !== null) {
-        // Round down the date to the nearest hour
-        date.setMinutes(0, 0, 0);
+        // eslint-disable-next-line max-depth
+        if (roundDownToNearestHour) {
+          // Round down the date to the nearest hour
+          date.setMinutes(0, 0, 0);
+        }
```

Comment:
Would it be simpler to move the rounding down to the calling code that wants it rather than introducing a flag onto this function?

Reply:
We would have to then do `date.setMinutes(0, 0, 0)` twice - once in v2 and once in v3 before calling this function, so it made sense to us to centralize it here especially since date logic/calculations can get tricky.

Reply:
Makes sense, I think there's some refactoring that could be done here to improve the code so we don't have to do this with a flag, but that's just a result of these two features sharing a similar function that creates the awkward code and it's out of scope for this change. I think the reuse is still worthwhile even with the added flag so let's go ahead with this and revisit in the future.